packages/typegpu/src/core/texture/texture.ts

```diff
@@ -0,0 +1,546 @@
+    invariant(!!type, `Unsupported storage texture format: ${format}`);
+
+    return new TgpuBindableStorageTextureImpl(params ?? {}, access, this);
+  }
+
+  asReadonly(
+    params?: TextureViewParams<
+      StorageTextureDimension,
+      StorageTextureTexelFormat
+    >,
+  ) {
+    // biome-ignore lint/suspicious/noExplicitAny: <too much type wrangling>
+    return this._asStorage(params, 'readonly') as any;
+  }
+
+  asWriteonly(
+    params?: TextureViewParams<
+      StorageTextureDimension,
+      StorageTextureTexelFormat
+    >,
+  ) {
+    // biome-ignore lint/suspicious/noExplicitAny: <too much type wrangling>
+    return this._asStorage(params, 'writeonly') as any;
+  }
+
+  asMutable(
+    params?: TextureViewParams<
+      StorageTextureDimension,
+      StorageTextureTexelFormat
+    >,
+  ) {
+    // biome-ignore lint/suspicious/noExplicitAny: <too much type wrangling>
+    return this._asStorage(params, 'mutable') as any;
+  }
+
+  asSampled(
+    params?: TextureViewParams<GPUTextureViewDimension, GPUTextureFormat>,
+    // biome-ignore lint/suspicious/noExplicitAny: <too much type wrangling>
+  ): any {
+    if (!this.usableAsSampled) {
+      throw new Error('Unusable as sampled');
+    }
+
+    const format = params?.format ?? this.props.format;
+    const type = texelFormatToDataType[format as keyof TexelFormatToDataType];
+
+    if (!type) {
+      throw new Error(`Unsupported storage texture format: ${format}`);
+    }
+
+    return new TgpuBindableSampledTextureImpl(params, 'readonly', this);
+  }
+
+  destroy() {
+    if (this._destroyed) {
+      return;
+    }
+    this._destroyed = true;
+    this._texture?.destroy();
+  }
+}
+
+const dimensionToCodeMap = {
+  '1d': '1d',
+  '2d': '2d',
+  '2d-array': '2d_array',
+  cube: 'cube',
+  'cube-array': 'cube_array',
+  '3d': '3d',
+} satisfies Record<GPUTextureViewDimension, string>;
+
+class TgpuBindableStorageTextureImpl
+  implements TgpuStorageTexture, TgpuStorageTexture_INTERNAL
+{
+  public readonly resourceType = 'texture-storage-view';
+  public readonly texelDataType: TexelData;
+  public readonly dimension: StorageTextureDimension;
+  /**
+   * TODO: Make private on resolution ctx refactor.
+   */
+  public readonly format: StorageTextureTexelFormat;
+
+  private _view: GPUTextureView | undefined;
+
+  constructor(
+    props:
+      | TextureViewParams<StorageTextureDimension, StorageTextureTexelFormat>
+      | undefined,
+    public readonly access: StorageTextureAccess,
+    private readonly _texture: TgpuTexture<TextureProps> & TgpuTexture_INTERNAL,
+  ) {
+    this.dimension = props?.dimension ?? _texture.props.dimension ?? '2d';
+    this.format =
+      props?.format ?? (_texture.props.format as StorageTextureTexelFormat);
+    this.texelDataType = texelFormatToDataType[this.format];
+  }
+
+  get label(): string | undefined {
+    return this._texture.label;
+  }
+
+  $name(label: string): this {
```

Comment:
I would actually like the naming aspect to be hidden from the user, since they are actually naming the texture resource underneath. This method will be called internally by `$uses()`